R/input-select.R

```diff
@@ -224,7 +228,7 @@ selectizeIt <- function(inputId, select, options, nonempty = FALSE) {
       type = 'application/json',
       `data-for` = inputId, `data-nonempty` = if (nonempty) '',
       `data-eval` = if (length(res$eval)) HTML(toJSON(res$eval)),
-      if (length(res$options)) HTML(toJSON(res$options)) else '{}'
+      HTML(toJSON(res$options))
```

Comment:
Why this change?

Reply:
Oh, I see, is it because `res$options` will always have positive length now?

Reply:
I changed this when I was trying a different method for fixing things, but it ended up being independent from the other stuff in this PR. Still, it simplified the code and I thought it was a good idea so I decided to keep it.

The `else` case of the original if-else was (apparently) to handle the case where `res$options` was either an empty list or `NULL` -- the `NULL` would get translated to a `null` in JSON. But now, with the code earlier in this function, if the incoming options were NULL, it gets turned into an empty named list, which then gets translated to `{}` in JSON. So the special case handling isn't necessary here.

Reply:
Although, now that you mention it, I see one potential issue: if `options` was an empty unnamed list, then the `toJSON()` would convert it to `[]` instead of `{}`. So I'll change the code above so that instead of `if (is.null(options))`, it will be `if (length(options) == 0)`.

Reply:
Oh, but in the case of an empty list or NULL, it would get turned into a named list anyway by the next `if` statement, where it adds the accessibility plugin to `options$plugins`.